payments-core/src/main/java/com/stripe/android/payments/core/authentication/DefaultPaymentAuthenticatorRegistry.kt

```diff
@@ -74,8 +88,38 @@ internal class DefaultPaymentAuthenticatorRegistry @Inject internal constructor(
 
     companion object {
         /**
-         * Create an instance of [PaymentAuthenticatorRegistry] with dagger.
+         * Host weak reference of all [DefaultPaymentAuthenticatorRegistry] instances created,
+         * the [Activity]s and [ViewModel]s started by the [PaymentAuthenticator]s hosted by the
+         * registry will need to access this instance to inject their dependencies.
+         * The reference will be garbage collected once it's no longer held by any client.
+         */
+        private val staticCache: MutableSet<DefaultPaymentAuthenticatorRegistry> =
+            Collections.newSetFromMap(WeakHashMap())
```

Comment:
Do we need to add this just to make `Stripe3ds2TransactionViewModel` injectable? I'm worried about the complexity cost we're introducing for something that will be used once.

Reply:
This is pretty similar to the singleton patterns used in [Terminal](https://git.corp.stripe.com/stripe-internal/android/blob/9740a311a47f564441c05f5c02aec4d063aa85c0/terminalsdk/core/src/main/kotlin/com/stripe/stripeterminal/Terminal.kt#L702) API, the difference is that we're hosting multiple weak instances instead of one strong instance here. 

Also as discussed offline, as a weak registry, as long as the public API is no longer hold by the user, it will be garbage collected, so we don't bear the responsibility to clean up this cache manually.